homeassistant/components/qwikswitch.py

```diff
@@ -74,129 +77,107 @@ def name(self):
     @property
     def is_on(self):
         """Check if device is on (non-zero)."""
-        return self._qsusb[self._id, 1] > 0
+        return self._qsusb[self.qsid, 1] > 0
 
-    def turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs):
         """Turn the device on."""
         new = kwargs.get(ATTR_BRIGHTNESS, 255)
-        self._qsusb.set_value(self._id, new)
+        self._qsusb.set_value(self.qsid, new)
 
-    @asyncio.coroutine
-    def async_turn_on(self, **kwargs):
-        """Turn the device on."""
-        new = kwargs.get(ATTR_BRIGHTNESS, 255)
-        self._qsusb.set_value(self._id, new)
-
-    def turn_off(self, **kwargs):  # pylint: disable=unused-argument
+    async def async_turn_off(self, **_):
         """Turn the device off."""
-        self._qsusb.set_value(self._id, 0)
-
-    @asyncio.coroutine
-    def async_turn_off(self, **kwargs):  # pylint: disable=unused-argument
-        """Turn the device off."""
-        self._qsusb.set_value(self._id, 0)
-
-
-class QSSwitch(QSToggleEntity, SwitchDevice):
-    """Switch based on a Qwikswitch relay module."""
+        self._qsusb.set_value(self.qsid, 0)
 
-    pass
-
-
-class QSLight(QSToggleEntity, Light):
-    """Light based on a Qwikswitch relay/dimmer module."""
-
-    @property
-    def brightness(self):
-        """Return the brightness of this light (0-255)."""
-        return self._qsusb[self._id, 1] if self._dim else None
-
-    @property
-    def supported_features(self):
-        """Flag supported features."""
-        return SUPPORT_BRIGHTNESS if self._dim else None
+    def async_added_to_hass(self):
+        """Listen for updates from QSUSb via dispatcher."""
+        # hass and schedule_update_ha_state is part of Entity/ToggleEntity
+        # pylint: disable=no-member
+        self.hass.helpers.dispatcher.async_dispatcher_connect(
+            self.qsid, lambda _=None: self.schedule_update_ha_state)
```

Comment:
You don't need the lambda.

 Use `async_schedule_update_ha_state` instead.

Reply:
Ok for async_schedule_update, but I prefer to have the lambda, 

There are two `async_dispatcher_send` calls, one includes a packet as parameter and one not. It should be safe that no switch or light entities will get called with the packet parameter, but I might not have 100% control over this (One scenario is where the devices list removes a device)

Reply:
I don't understand the scenario where this could problem would surface. If you want to keep it super safe, please define a regular method that accepts `*args` and calls `self.async_schedule_update_ha_state`.